py/farm_ng/core/event_service_recorder.py

```diff
@@ -75,15 +75,13 @@ def __init__(self, service_name: str, config_list: EventServiceConfigList) -> No
         self._clients: dict[str, EventClient] = {}
 
         # find the recorder config and create the clients
-        # TODO: why do not pass directly the config instead of the config list?
+        # Here we reuse a config for the subscriptions, from the recorder_config
+        # and create the clients for the other services (with port names)
```

Comment:
passing the list can be confusing and error prone. I'd go for passing two arguments in the constructor `recorder_config` (a config ds with subscriptions only for later subscribe) and then the `config_list` containing the existing services with the  info needed to create the client (host, port, etc). I also feel that having a single `EventServiceConfig` that can be both for creating client and defining service is a bit counter intuitive to interpret.

Reply:
This is designed to make the config.json structure easy to edit, with a single flat list of configuration sections.  Also, services will also have clients, so having a single EventServiceConfig that is either describing just a service, or a client, or a service and a client allows us to express this.

Reply:
We'll revisit useability before we make an official release.